docs/reference/transfers.md

```diff
@@ -3,14 +3,150 @@
 A `transfer` is an immutable record of a financial transaction between
 two accounts.
 
+In TigerBeetle, financial transactions are called "transfers" instead of "transactions" because
+the latter term is heavily overloaded in the context of databases.
+
+### Updates
+
+Transfers *cannot be modified* after creation.
+
+
+## Modes
+
+Fields used by each mode of transfer:
+
+| Field                         | Single-Phase | Pending  | Post-Pending | Void-Pending |
+| ----------------------------- | ------------ | -------- | ------------ | ------------ |
+| `id`                          | required     | required | required     | required     |
+| `debit_account_id`            | required     | required | optional     | optional     |
+| `credit_account_id`           | required     | required | optional     | optional     |
+| `user_data`                   | optional     | optional | optional     | optional     |
+| `reserved`                    | none         | none     | none         | none         |
+| `pending_id`                  | none         | none     | required     | required     |
+| `timeout`                     | none         | required | none         | none         |
+| `ledger`                      | required     | required | optional     | optional     |
+| `code`                        | required     | required | optional     | optional     |
+| `flags.linked`                | optional     | optional | optional     | optional     |
+| `flags.pending`               | false        | true     | false        | false        |
+| `flags.post_pending_transfer` | false        | false    | true         | false        |
+| `flags.void_pending_transfer` | false        | false    | false        | true         |
+| `amount`                      | required     | required | optional     | optional     |
+
 TigerBeetle uses the same data structures internally and
 externally. This means that sometimes you need to set temporary values
 for fields that TigerBeetle, not you (the user), are responsible.
 
-### Updates
+### Single-Phase Transfer
+
+Single-phase transfers post funds to accounts immediately when they are created.
+
+### Pending Transfer
+
+A pending transfer (denoted by [`flags.pending`](#flagspending) followed by a post-pending transfer,
+void-pending transfer, or a timeout is called a "two-phase transfer".
```

Comment:
You have a section dedicated to Single-phase transfer which makes sense but then the two-phase transfer doesn't really have a section which feels asymmetric. On the other hand I get that these sections just refer to the type of the transfer.

Reply:
This is a good point. I was thinking of it in terms of the different transfer types — one section each. But I agree; I'll group those within a "Two-Phase Transfer" section instead.